### DISCUSSION TITLE
`useQueryState` overload that returns all the URLSearchParams?

### DISCUSSION THREAD

**Discussion by ChapterSevenSeeds** [47ng/nuqs#609] [Q&A] [answered]
Hello. I am working on a page in a Next.js app (with the App Router) that stores a minified version of a nasty search query into the URL query params. This query undergoes frequent changes and I need the client to react immediately to these changes. Since learning that the Next.js App Router doesn't support shallow routing, I have stumbled across Nuqs and want to give it a try. However, to fit my use case, I would need a version of the `useQueryState` hook that I can call with no arguments that gives me the entirety of the current URL search params, similar to the functionality of the `useSearchParams` exported by `react-router-dom`. Is this possible with Nuqs?

**Comment by franky47** [comment, marked as answer]
It's actually just a Next.js API, [`useSearchParams`](https://nextjs.org/docs/app/api-reference/functions/use-search-params). nuqs uses it internally, and adds a few things on top (mapping to one particular key, type-safety via parsing, and shallow/non-shallow updates).

FYI, Next.js (app router) does support shallow updates now (on search params only), using the [History API](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#using-the-native-history-api). That's what nuqs does under the hood.

**Reply by ChapterSevenSeeds** [reply]
So I was using the History API to imitate the shallow routing but Next.js wasn't picking up on any of the changes I was making to the query params. Previously I was doing `window.history.pushState(window.history.state, "", href)`. However, I just learned that all I needed to do was pass in `null` for the first argument. Now it works as intended. Thanks!
